src/lock/parse.rs

```diff
@@ -0,0 +1,230 @@
+        )
+        .to_string(),
+    })
+}
+
+// Note: technically this function cannot fail, but we want to keep symetry with
+// parse_git_line()
+pub fn parse_simple_line(line: &str) -> Result<SimpleDependency, Error> {
+    let version = parse_simple_version(&line);
+    let name = parse_simple_name(&line);
+    Ok(SimpleDependency {
+        line: line.to_string(),
+        name,
+        version,
+    })
+}
+
+fn parse_simple_name(line: &str) -> String {
+    let words: Vec<_> = line.split("==").collect();
+    let dep_name = words[0];
+    dep_name.trim().to_string()
+}
+
+fn parse_simple_version(line: &str) -> VersionSpec {
+    let equal_index = line
+        .find("==")
+        .unwrap_or_else(|| panic!("'{}' should contain '=='", line));
+    let mut start = equal_index + 2;
+
+    // Advance start until the next non-blank char
+    loop {
+        let chr = &line.chars().nth(start);
+        if chr.is_none() {
+            break;
+        }
+        let chr = chr.unwrap();
+        if !chr.is_ascii_whitespace() {
+            break;
+        }
+        start += 1;
+    }
+
+    // Advance `end` until:
+    // * the next blank char,
+    // * the end of the line,
+    // * the backslash character (before a line continuation)
+    // * the hash character (before a comment)
+    let mut end = start;
+    loop {
+        let chr = &line.chars().nth(end);
+        if chr.is_none() {
+            break;
+        }
+        let chr = chr.unwrap();
+        if chr.is_ascii_whitespace() || (chr == '\\') || (chr == '#') {
+            break;
+        }
+        end += 1;
+    }
+
+    let version = &line[start..end];
+
+    VersionSpec {
+        start,
+        end,
+        value: version.to_string(),
+    }
+}
+
+pub fn parse_git_line(line: &str) -> Result<GitDependency, Error> {
+    let name = parse_git_name(line);
+    let git_ref = parse_git_ref(line)?;
+    Ok(GitDependency {
+        line: line.to_string(),
+        name,
+        git_ref,
+    })
+}
+
+fn parse_git_name(line: &str) -> String {
+    let parts: Vec<_> = line.rsplit("egg=").collect();
+    let dep_name = parts[0];
+    dep_name.trim().to_string()
+}
+
+fn parse_git_ref(line: &str) -> Result<VersionSpec, Error> {
+    let chunks: Vec<_> = line.rsplit('@').collect();
+    // chunks is [git, foo:com:bar/baz, abce64#egg=bar]
+    let after_at = chunks
+        .first()
+        .unwrap_or_else(|| panic!("'{}' should contain '@'", line));
+
+    let chunks: Vec<_> = after_at.split('#').collect();
+    // chunks is [abce64, egg=bar]
+    if chunks.len() != 2 {
+        return Err(Error::MalformedLock {
+            details: format!(
+                "expecting `<ref>#egg=<name>` after `@`, got '{}'",
+                after_at.trim_end()
+            ),
+        });
+    }
+    let value = chunks[0].to_string();
+    let start = line.len() - after_at.len();
+    let end = start + value.len();
+    Ok(VersionSpec { value, start, end })
+}
+#[cfg(test)]
+mod tests {
```

Comment:
I know this isn't PEP8 territory, but a newline before the module wouldn't be too much here =]